### ISSUE TITLE
🐛 [Firebase_auth] idToken is always changed when call getIdToken() on iOS device

### ISSUE THREAD

**Issue by palmmy2542** [firebase/flutterfire#11428]
## Bug report

**Describe the bug**
As I mention above. I found this issue only iOS. 

### Steps to reproduce

I try this step on demo in this repo

1. Run app on iOS device
2. Create a button to call `getIdToken()` or `getIdToken(false)`

Result:

issuedAtTime, expirationTime, token are always changed after click a button

### Expected behavior

issuedAtTime, expirationTime, token should not be changed til token expired.

### Sample project

https://github.com/palmmy2542/flutterfire
---

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

<img width="633" alt="image" src="https://github.com/firebase/flutterfire/assets/54408097/b054eeaa-1028-4ab1-aaa2-06f68f485644">
<img width="709" alt="image" src="https://github.com/firebase/flutterfire/assets/54408097/dcd66902-6302-467c-b649-8775f1faa59c">

</details>

---

### Flutter dependencies

Run `flutter pub deps -- --style=compact` and paste the output below:

<details><summary>Click To Expand</summary>

```
Dart SDK 3.0.6
Flutter SDK 3.10.6
flutterfire_workspace 0.0.0

dev dependencies:
- melos 3.1.1 [ansi_styles args cli_launcher cli_util collection conventional_commit file glob graphs http meta mustache_template path platform pool prompts pub_semver pub_updater pubspec string_scanner yaml yaml_edit]

transitive dependencies:
- ansi_styles 0.3.2+1
- args 2.4.2
- async 2.11.0 [collection meta]
- boolean_selector 2.1.1 [source_span string_scanner]
- charcode 1.3.1
- cli_launcher 0.3.1 [path yaml]
- cli_util 0.4.0 [meta path]
- collection 1.18.0
- conventional_commit 0.6.0+1
- file 6.1.4 [meta path]
- glob 2.1.2 [async collection file path string_scanner]
- graphs 2.3.1 [collection]
- http 1.1.0 [async http_parser meta]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- io 1.0.4 [meta path string_scanner]
- json_annotation 4.8.1 [meta]
- matcher 0.12.16 [async meta stack_trace term_glyph test_api]
- meta 1.9.1
- mustache_template 2.0.0
- path 1.8.3
- platform 3.1.0
- pool 1.5.1 [async stack_trace]
- process 4.2.4 [file path platform]
- prompts 2.0.0 [charcode io]
- pub_semver 2.1.4 [collection meta]
- pub_updater 0.3.1 [http json_annotation process pub_semver]
- pubspec 2.3.0 [path pub_semver yaml uri]
- quiver 3.2.1 [matcher]
- source_span 1.10.0 [collection path term_glyph]
- stack_trace 1.11.1 [path]
- stream_channel 2.1.2 [async]
- string_scanner 1.2.0 [source_span]
- term_glyph 1.2.1
- test_api 0.6.1 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- typed_data 1.3.2 [collection]
- uri 1.0.0 [matcher quiver]
- yaml 3.1.2 [collection source_span string_scanner]
- yaml_edit 2.1.1 [collection meta source_span yaml]
```

</details>

---

**Comment by danagbemava-nc** [comment]
I see same when I run the plugin example app and use the following code on the profile page to call the getIdTokenResult(). I not sure if this is an issue with the plugin or the native SDK. Labeling

```dart
TextButton(
                        onPressed: () async {
                          final user = FirebaseAuth.instance.currentUser;
                          final idToken = await user?.getIdTokenResult();

                          // print(user);
                          print(idToken);
                        },
                        child: Text('Get token'),
                      )
```

```
[3]: Nexus (00008020-001875E83A38002E)
Please choose one (or "q" to quit): 3
Launching lib/main.dart on Nexus in debug mode...
Warning: Missing build name (CFBundleShortVersionString).
Warning: Missing build number (CFBundleVersion).
Action Required: You must set a build name and number in the pubspec.yaml file version field before submitting to the App Store.
Automatically signing iOS for device deployment using specified development team in Xcode project: X8NNQ9CYL2
Running Xcode build...                                                  
 └─Compiling, linking and signing...                         6.3s
Xcode build done.                                           36.6s
(lldb) 2023-08-08 13:11:17.312704+0000 Runner[843:95249] [VERBOSE-2:FlutterDarwinContextMetalImpeller.mm(35)] Using the Impeller rendering backend.
10.12.0 - [FirebaseCore][I-COR000005] No app has been configured yet.
10.12.0 - [FirebaseCore][I-COR000008] The project's Bundle ID is inconsistent with either the Bundle ID in 'GoogleService-Info.plist', or the Bundle ID in the options if you are using a customized options. To ensure that everything can be configured correctly, you may need to make the Bundle IDs consistent. To continue with this plist file, you may change your app's bundle identifier to 'io.flutter.plugins.firebase.auth.example'. Or you can download a new configuration file that matches your bundle identifier from https://console.firebase.google.com/ and replace the current one.
10.12.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
Installing and launching...                                        54.0s
Syncing files to device Nexus...                                   256ms

Flutter run key commands.
r Hot reload. 🔥🔥🔥
R Hot restart.
h List all available interactive commands.
d Detach (terminate "flutter run" but leave application running).
c Clear the screen
q Quit (terminate the application on the device).

A Dart VM Service on Nexus is available at: http://127.0.0.1:63442/jqONzV0za1Q=/
The Flutter DevTools debugger and profiler on Nexus is available at: http://127.0.0.1:9101?uri=http://127.0.0.1:63442/jqONzV0za1Q=/
flutter: IdTokenResult(authTime: 2023-08-08 12:15:24.000, claims: {iat: 1691500311, email: payapp54@gmail.com, firebase: {identities: {email: [payapp54@gmail.com]}, sign_in_provider: password}, email_verified: true, sub: NviciwXJrjXbznzWDgSbdwT3Tt63, aud: flutterfire-e2e-tests, auth_time: 1691496924, user_id: NviciwXJrjXbznzWDgSbdwT3Tt63, exp: 1691503911, name: Nexus, iss: https://securetoken.google.com/flutterfire-e2e-tests}, expirationTime: 2023-08-08 14:11:51.000, issuedAtTime: 2023-08-08 13:11:51.000, signInProvider: password, token: eyJhbGciOiJSUzI1NiIsImtpZCI6ImNmM2I1YWRhM2NhMzkxNTQ4ZDM1OTJiMzU5MjkyM2UzNjAxMmI5MTQiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiTmV4dXMiLCJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXVkIjoiZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXV0aF90aW1lIjoxNjkxNDk2OTI0LCJ1c2VyX2lkIjoiTnZpY2l3WEpyalhiem56V0RnU2Jkd1QzVHQ2MyIsInN1YiI6Ik52aWNpd1hKcmpYYnpueldEZ1NiZHdUM1R0NjMiLCJpYXQiOjE2OTE1MDAzMTEsImV4cCI6MTY5MTUwMzkxMSwiZW1haWwiOiJwYXlhcHA1NEBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiZmlyZWJhc2UiOnsiaWRlbnRpdGllcyI6eyJlbWFpbCI6WyJwYXlhcHA1NEBnbWFpbC5jb20iXX0sInNpZ25faW5fcHJvdmlkZXIiOiJwYXNzd29yZCJ9fQ.IUwPDBOlX3Mr09g5YNp6Xdpvu29NQmTfrqxs8zZtXKHS6OviyddBP2FDpM09B627Z_AGGYcR7XBwJ3271OVudVoD-BiYP_QWJAe24kTfLc9a2-FWumS-o3jVuCx04F-nxPBwPWkCyOFtCNrFsdxEBpA2qs3D2dK8VOQmftEHcgDEyOEqvdg2yYmaw3ES0z634T0C3L1pRtRSOshrtnRyM-9Agr8EcftdOy4znH97QcROJBQ9WsVDlxXwNRSYD3MSAIHaa-Vyp2rZKgmM6FI6utivRQLxYdUFNKgoeUEi0bJpWKwQlfPltUDvQh17Ai0budc0wDfwtWPpvoib3BwzaA)
flutter: IdTokenResult(authTime: 2023-08-08 12:15:24.000, claims: {iat: 1691500315, email: payapp54@gmail.com, firebase: {identities: {email: [payapp54@gmail.com]}, sign_in_provider: password}, email_verified: true, sub: NviciwXJrjXbznzWDgSbdwT3Tt63, aud: flutterfire-e2e-tests, auth_time: 1691496924, user_id: NviciwXJrjXbznzWDgSbdwT3Tt63, exp: 1691503915, name: Nexus, iss: https://securetoken.google.com/flutterfire-e2e-tests}, expirationTime: 2023-08-08 14:11:55.000, issuedAtTime: 2023-08-08 13:11:55.000, signInProvider: password, token: eyJhbGciOiJSUzI1NiIsImtpZCI6ImNmM2I1YWRhM2NhMzkxNTQ4ZDM1OTJiMzU5MjkyM2UzNjAxMmI5MTQiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiTmV4dXMiLCJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXVkIjoiZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXV0aF90aW1lIjoxNjkxNDk2OTI0LCJ1c2VyX2lkIjoiTnZpY2l3WEpyalhiem56V0RnU2Jkd1QzVHQ2MyIsInN1YiI6Ik52aWNpd1hKcmpYYnpueldEZ1NiZHdUM1R0NjMiLCJpYXQiOjE2OTE1MDAzMTUsImV4cCI6MTY5MTUwMzkxNSwiZW1haWwiOiJwYXlhcHA1NEBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiZmlyZWJhc2UiOnsiaWRlbnRpdGllcyI6eyJlbWFpbCI6WyJwYXlhcHA1NEBnbWFpbC5jb20iXX0sInNpZ25faW5fcHJvdmlkZXIiOiJwYXNzd29yZCJ9fQ.RoW3k_8tPp9Jc-4oheeo6LVMUs7XbBVx1y7Qjulu4rMEdgWRIDfso3lQpC2jbMAOSSp7Q1ZTxGPbnUgIDG3MA5nIPQqrfPaFoTj2Wjbg7AfrqLU7hbBgidq7mnJ7GdregU9upOQekMePSFPtMk6zGCnBWDuul8t-aIW38gRQWeWKjEqeZjrrUkfOAQ3FcCxu2xpjL9rFbwLTIxu0vy0DBEzcyjvaDtR-LBMT2pYyl-z95LVFqQzUWQ_GXbBDb6xxr2M9Sxs5GUJOk_7ro8w7C2UcFnoC35U5nXb0EMzN9ua8QgweNhhA0eFJEIFmUBHdaVJLaX2OlJSzakfFXPPHbw)
flutter: IdTokenResult(authTime: 2023-08-08 12:15:24.000, claims: {iat: 1691500319, email: payapp54@gmail.com, firebase: {identities: {email: [payapp54@gmail.com]}, sign_in_provider: password}, email_verified: true, sub: NviciwXJrjXbznzWDgSbdwT3Tt63, aud: flutterfire-e2e-tests, auth_time: 1691496924, user_id: NviciwXJrjXbznzWDgSbdwT3Tt63, exp: 1691503919, name: Nexus, iss: https://securetoken.google.com/flutterfire-e2e-tests}, expirationTime: 2023-08-08 14:11:59.000, issuedAtTime: 2023-08-08 13:11:59.000, signInProvider: password, token: eyJhbGciOiJSUzI1NiIsImtpZCI6ImNmM2I1YWRhM2NhMzkxNTQ4ZDM1OTJiMzU5MjkyM2UzNjAxMmI5MTQiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiTmV4dXMiLCJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXVkIjoiZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXV0aF90aW1lIjoxNjkxNDk2OTI0LCJ1c2VyX2lkIjoiTnZpY2l3WEpyalhiem56V0RnU2Jkd1QzVHQ2MyIsInN1YiI6Ik52aWNpd1hKcmpYYnpueldEZ1NiZHdUM1R0NjMiLCJpYXQiOjE2OTE1MDAzMTksImV4cCI6MTY5MTUwMzkxOSwiZW1haWwiOiJwYXlhcHA1NEBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiZmlyZWJhc2UiOnsiaWRlbnRpdGllcyI6eyJlbWFpbCI6WyJwYXlhcHA1NEBnbWFpbC5jb20iXX0sInNpZ25faW5fcHJvdmlkZXIiOiJwYXNzd29yZCJ9fQ.MCdbASAEQlohOo2eXVn7DIw4bhDYyyi-sRH3Si8cUHbNl8C7EUs7WLDJdExjsJcMHTXzqRh80A1Z7jMU2eN1quDusojbpaCf-MX_8rbfTzHOkWrC0sscbn4otozoC7PxkcbnhYEh_qJiy-W8KYsVuy0Z8MsU0QGBhntGmvKhG7wUyEQCr9ntLg449zwbopJvMUOZzARgI8seR7CdaxthrQmhUa0nEtmZDI3S9EdrVCd3q1D-yXPCMh-P7JTd-mCShTXN_Yq0MHag0lvPgxaCqB7-73bAj3ektnsw6pkDqYd36gs04iuWMjWxeb-5ZwTzgAcKrXYRtbQJ6x2EHVcDQQ)
flutter: IdTokenResult(authTime: 2023-08-08 12:15:24.000, claims: {iat: 1691500321, email: payapp54@gmail.com, firebase: {identities: {email: [payapp54@gmail.com]}, sign_in_provider: password}, email_verified: true, sub: NviciwXJrjXbznzWDgSbdwT3Tt63, aud: flutterfire-e2e-tests, auth_time: 1691496924, user_id: NviciwXJrjXbznzWDgSbdwT3Tt63, exp: 1691503921, name: Nexus, iss: https://securetoken.google.com/flutterfire-e2e-tests}, expirationTime: 2023-08-08 14:12:01.000, issuedAtTime: 2023-08-08 13:12:01.000, signInProvider: password, token: eyJhbGciOiJSUzI1NiIsImtpZCI6ImNmM2I1YWRhM2NhMzkxNTQ4ZDM1OTJiMzU5MjkyM2UzNjAxMmI5MTQiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiTmV4dXMiLCJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXVkIjoiZmx1dHRlcmZpcmUtZTJlLXRlc3RzIiwiYXV0aF90aW1lIjoxNjkxNDk2OTI0LCJ1c2VyX2lkIjoiTnZpY2l3WEpyalhiem56V0RnU2Jkd1QzVHQ2MyIsInN1YiI6Ik52aWNpd1hKcmpYYnpueldEZ1NiZHdUM1R0NjMiLCJpYXQiOjE2OTE1MDAzMjEsImV4cCI6MTY5MTUwMzkyMSwiZW1haWwiOiJwYXlhcHA1NEBnbWFpbC5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiZmlyZWJhc2UiOnsiaWRlbnRpdGllcyI6eyJlbWFpbCI6WyJwYXlhcHA1NEBnbWFpbC5jb20iXX0sInNpZ25faW5fcHJvdmlkZXIiOiJwYXNzd29yZCJ9fQ.T7s_FMU4V7QsnbEE8ULYYHGGUaN8FcEGzSOg6Brr-XCM6OkbFnW9b3X-Fd6rrLcksXQO-GhFjXOwjZ54F6zQCWNwKN28xM3n2UuyxcKvT2CAuYLFsicj2wBxoZweuXfrTGw8jz8hg6GtQDMCB4sxOywq98-90F5K8rjbHGtUn6MtS6WrqiayHkc3K9MaLd-HNsAMVKphNW9MTO_wJ7oVAS_YmDKAf0WhmUhFUKmRUdzuRiHuSVRuOb3yUe55ZG-P00ilpkidgZY8Cn4M7o7NxY2PW-QWpvMhsfJg9WBl9AyncEnOtaKskWtdwzMEdoIWmhXEg6GBJwUwbKUFD3HACg)
Application finished.
```

**Comment by HatsuneMikuV** [comment]
I have same issues
token are always changed when getIdToken(false)

**Comment by hnvn** [comment]
Same issue, this behavior only occurs on iOS, it is normal on Android.

**Comment by limenote135** [comment]
Is it still happening? I'm not sure, but it doesn't seem to be happening now.

**Comment by JishnuM** [comment]
Believe this is fixed by https://github.com/firebase/flutterfire/pull/11464 included in `4.7.3`

**Comment by palmmy2542** [comment]
Thank. it was solved !
